### ISSUE TITLE
Webpack compile error while loading package in Electron App

### ISSUE THREAD

**Issue by prabir-vora** [GetStream/stream-chat-react#1675]
**Describe the bug**

I am trying to use stream-chat-react in my Expo Electron app. However, I am not able to get my app to compile since Webpack is throwing some errors while loading this package. You can find the screenshot below. 

**SCREEN SHOT OF BUILD ERROR**

<img width="1272" alt="Screen Shot 2022-07-14 at 11 10 45 PM" src="https://user-images.githubusercontent.com/46608269/179142424-84b909e4-c5fc-428e-bfaa-f9e9e422f0f1.png">

**CODE FILES**

**webpack.config.js**

```
import dotenv from 'dotenv';
const path = require('path');

const createExpoWebpackConfigAsync = require('@expo/webpack-config');
module.exports = async function (env, argv) {
    const config = await createExpoWebpackConfigAsync(env, argv);
    config.resolve.alias['react-native-webview'] = 'react-native-web-webview';
    // config.resolve.alias['react/jsx-runtime'] = 'react/jsx-runtime.js';
    // config.resolve.alias['react/jsx-dev-runtime'] = 'react/jsx-dev-runtime.js';

    config.resolve.alias['react-native-webview'] = 'react-native-web-webview';

    config.resolve.alias['react'] = path.resolve('./node_modules/react');

    config.resolve.alias['FroalaEditor'] = 'froala_editor.min.js/froala_editor.pkgd.min.js';

    config.resolve.modules = ['../node_modules/froala-editor/js', 'node_modules'];

    config.plugins.push(
        new webpack.ProvidePlugin({
            FroalaEditor: 'froala_editor.min.js/froala_editor.pkgd.min.js',
        })
    );

    config.plugins.push(
        new webpack.EnvironmentPlugin({
            NODE_ENV: 'production',
            ...dotenv.config().parsed, // <-- this line
        })
    );

    return config;
};

```

**electron-webpack.js**

Used to whitelist react modules

```
const { withExpoAdapter } = require('@expo/electron-adapter');

ELECTRON_DISABLE_SECURITY_WARNINGS = 1;

module.exports = withExpoAdapter({
    projectRoot: __dirname,
    whiteListedModules: [
        'stream-chat-react',
        'stream-chat',
...others
    ],
    // Provide any overrides for electron-webpack: https://github.com/electron-userland/electron-webpack/blob/master/docs/en/configuration.md
});
```

**babel.config.js**

```
module.exports = function(api) {
  api.cache(true);
  return {
    presets: ['babel-preset-expo'],
  };
};

```

**tsconfig.json**

```

{
  "compilerOptions": {
    "allowSyntheticDefaultImports": true,
    "jsx": "react-native",
    "lib": ["dom", "esnext"],
    "moduleResolution": "node",
    "noEmit": true,
    "skipLibCheck": true,
    "resolveJsonModule": true,
    "strict": false
  }
}


```

**Package version**
(Ones I think may be relevant)

- stream-chat-react: "^9.1.4"
- stream-chat-js: "^6.7.3",
- typescript: "^4.4.3"
- @expo/electron-adapter: "^0.0.55"
- @expo/webpack-config: "~0.12.45"
- @babel/core: "~7.9.0"

**CONTEXT**

OS: macOS
Node version: v16.15.1
webpack version: 4.43.0

Need help asap. If a fix requires time I would appreciate it if someone can guide me on what could be done from my end to fix this.

**Comment by oliverlaz** [comment]
hi @prabir-vora, thanks for reaching out. Can you please submit a minimal reproduction example for this issue so we can check it out more closely? In meantime, it might be worth checking your babel configuration. We assume it needs some tweaks on your side.

**Comment by prabir-vora** [comment]
@oliverlaz Thanks for reaching out. I have created a minimal reproduction example [https://github.com/prabir-vora/stream-chat-electron](url). 

Please note my webpack version at compile time is 4.43.0.

I did some more debugging and seems like upgrading my webpack to v5 gets rid of this issue but it causes other errors since electron-webpack doesn't support webpack v5 yet. 

Once again I need help at the earliest. Thanks!

**Comment by prabir-vora** [comment]
@oliverlaz is the repo I created sufficient to replicate the issue on your end? Please let me know if anything additional is required from my end.

**Comment by oliverlaz** [comment]
hi @prabir-vora, we looked at this problem and managed to find a hacky solution to the described problem.
In order to make webpack/babel compile the problematic code, please apply these changes:

`yarn add --dev @babel/preset-env @babel/plugin-proposal-export-namespace-from`

```javascript
// babel.config.js:
module.exports = function(api) {
  api.cache(true);
  return {
    presets: ['babel-preset-expo', ["@babel/preset-env", {
      targets: {
        electron: 19,
      },
      loose: true,
    }]],
    plugins: ['@babel/plugin-proposal-export-namespace-from'],
  };
};
```

```javascript
// electron/webpack.config.js:
const { withExpoWebpack } = require('@expo/electron-adapter');

module.exports = config => {
  const configuration = withExpoWebpack(config);
  for (const rule of configuration.module.rules) {
    if ('.js'.match(rule.test)) {
      // let babel-loader compile all source files
      rule.exclude = [];
    }
  }
  return configuration;
};
```

With this, the project compiles but there are other problems which are surfacing afterwards. This is a sign that the experimental `expo-electron` setup isn't ideal and further issues can be expected. We can't allocate more time for further research and assistance, as it's a very niche and experimental setup.

Our colleague @khushal87 already recommended you to use [electron-react](https://github.com/electron-react-boilerplate/electron-react-boilerplate) as a better alternative. Perhaps, you can give it a try.

**Comment by prabir-vora** [comment]
@oliverlaz Thanks for the help. I just got it to work with a different workaround than the one mentioned above. The above solution was causing my app to not compile at all.

All I did was simply modify the following line in htmlparser2 where the error is thrown

`export * as DomUtils from "domutils"` 

Change it to

`import * as DomUtils from "domutils"`
`export { DomUtils }`

No Babel config change was required. Thought I would post this in case someone else runs into this issue. I know it's only a temporary workaround but I'll take it for now
